src/qibolab/instruments/qm.py

```diff
@@ -65,15 +68,13 @@ def declare_output(self, threshold=None, iq_angle=None):
             self.sin = np.sin(iq_angle)
 
     def bake(self, config):
-        from qualang_tools.bakery import baking
-
         if self.baked is not None:
             raise_error(RuntimeError, f"Bake was already called for {self.pulse}.")
         # ! Only works for flux pulses that have zero Q waveform
 
         # Create the different baked sequences, each one corresponding to a different truncated duration
         # for t in range(self.pulse.duration + 1):
-        with baking(config, padding_method="right") as self.baked:
+        with baking(config.__dict__, padding_method="right") as self.baked:
```

Comment:
Here and below: `config` is supposed to be a `dataclass`, thus you can use
```suggestion
        with baking(asdict(config), padding_method="right") as self.baked:
```

`asdict()` is a function part of the `dataclasses` module:
```py
from dataclasses import asdict
```

There might be multiple reasons to advocate for it, and you can summarize just saying that is cleaner. But in order to show a practical example, consider that a `dataclass` field is not the same concept as an attribute, and `__dict__` can contain any dynamic added attribute, while fields are only those recognized by the decorator.

Reply:
Specifically for `baking()`, this function modifies the input dictionary, so it is not possible to use `asdict()` because it creates a copy of the field dictionary, while we want `baking` to modify the original `config` object. To explain in more detail
```py
config = {}
with baking(config, ...):
    ...
```
will result in the config having a new element like `config = {"pulses": {"baked_pulse0": ...}}`, which is added within the `baking()` call that is part of a dependency. If we use `asdict()` we need to make sure to update the original config after, but I believe this will complicate the code a bit more.

For all other places I use `__dict__`, it can be replaced to `asdict()` without problem, the only reason I did not do it is https://stackoverflow.com/questions/52229521/why-is-dataclasses-asdictobj-10x-slower-than-obj-dict
and because making a copy is not mandatory in our case. I understand both your arguments, but assuming proper usage, that is not dynamically adding arguments to `config`, `__dict__` will most likely not break. One possibility is to change to private `self._config` to make sure it is only modified within the driver, which we control.